### PR TITLE
Issue 2034

### PR DESCRIPTION
**Pull request by nmundar**
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs:

- [x] Target: for bug or doc fixes, target `master`; for new features, target `features`;

Unless your change is trivial documentation fix (e.g.,  a typo or reword of a small section) please:

- [ ] Make sure to include one or more tests for your change;
- [x] Add yourself to `AUTHORS`;
- [x] Add a new entry to `CHANGELOG.rst`
  * Choose any open position to avoid merge conflicts with other PRs.
  * Add a link to the issue you are fixing (if any) using RST syntax.
  * The pytest team likes to have people to acknowledged in the `CHANGELOG`, so please add a thank note to yourself ("Thanks @user for the PR") and a link to your GitHub profile. It may sound weird thanking yourself, but otherwise a maintainer would have to do it manually before or after merging instead of just using GitHub's merge button. This makes it easier on the maintainers to merge PRs.

### CODE REVIEW

File: _pytest/main.py [hunk 1]
```diff
@@ -190,7 +190,9 @@ def __getattr__(self, name):
 
 def compatproperty(name):
     def fget(self):
-        # deprecated - use pytest.name
+        import warnings
+        warnings.warn("compatproperty is deprecated. Use pytest.name",
```
Comment (nicoddemus) [inline _pytest/main.py]: This should be: `"compatproperty is deprecated. Use pytest.{0}".format(name)`

**Comment by nicoddemus** [comment]
Thanks!

**Comment by nicoddemus** [comment]
Also please rebase on `master`.

**Comment by nmundar** [comment]
Argh, sorry for the typo in the commit message!

**Comment by nicoddemus** [comment]
Feel free to amend it. I just noticed that the CHANGELOG message could also
mention that what's being deprecated is accessing item.Function,
item.Module, etc. If you amend the commit could you update the CHANGELOG as
well? If not, don't worry about it, I can do it later

**Comment by nmundar** [comment]
Please, if it's not a bother, update the changelog as you see fit, you have a better idea about wording. Don't want to mess up, the same with force pushing amended commits.

**Comment by nicoddemus** [comment]
Sure, I will do it before merging, thanks!

**Comment by nicoddemus** [comment]
Updated CHANGELOG and rebased on `master`. Forced push to the branch in your fork @nmundar, hope you don't mind.

**Pull request merged**